spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/dynamodb/DynamoDbAutoConfiguration.java

```diff
@@ -60,39 +70,59 @@ public class DynamoDbAutoConfiguration {
-				AwsRegionProvider regionProvider) throws IOException {
-			DaxProperties daxProperties = properties.getDax();
-
+		private software.amazon.dax.Configuration.Builder toAwsDaxConfiguration(DaxProperties daxProperties) {
 			PropertyMapper propertyMapper = PropertyMapper.get();
 			software.amazon.dax.Configuration.Builder configuration = software.amazon.dax.Configuration.builder();
 			propertyMapper.from(daxProperties.getIdleTimeoutMillis()).whenNonNull()
-					.to(configuration::idleTimeoutMillis);
+				.to(configuration::idleTimeoutMillis);
 			propertyMapper.from(daxProperties.getConnectionTtlMillis()).whenNonNull()
-					.to(configuration::connectionTtlMillis);
+				.to(configuration::connectionTtlMillis);
 			propertyMapper.from(daxProperties.getConnectTimeoutMillis()).whenNonNull()
-					.to(configuration::connectTimeoutMillis);
+				.to(configuration::connectTimeoutMillis);
 			propertyMapper.from(daxProperties.getRequestTimeoutMillis()).whenNonNull()
-					.to(configuration::requestTimeoutMillis);
+				.to(configuration::requestTimeoutMillis);
 			propertyMapper.from(daxProperties.getWriteRetries()).whenNonNull().to(configuration::writeRetries);
 			propertyMapper.from(daxProperties.getReadRetries()).whenNonNull().to(configuration::readRetries);
 			propertyMapper.from(daxProperties.getClusterUpdateIntervalMillis()).whenNonNull()
-					.to(configuration::clusterUpdateIntervalMillis);
+				.to(configuration::clusterUpdateIntervalMillis);
 			propertyMapper.from(daxProperties.getEndpointRefreshTimeoutMillis()).whenNonNull()
-					.to(configuration::endpointRefreshTimeoutMillis);
+				.to(configuration::endpointRefreshTimeoutMillis);
 			propertyMapper.from(daxProperties.getMaxConcurrency()).whenNonNull().to(configuration::maxConcurrency);
 			propertyMapper.from(daxProperties.getMaxPendingConnectionAcquires()).whenNonNull()
-					.to(configuration::maxPendingConnectionAcquires);
+				.to(configuration::maxPendingConnectionAcquires);
 			propertyMapper.from(daxProperties.getSkipHostNameVerification()).whenNonNull()
-					.to(configuration::skipHostNameVerification);
+				.to(configuration::skipHostNameVerification);
+
+			return configuration;
+		}
+
+		@ConditionalOnMissingBean
+		@Bean
+		public DynamoDbClient daxDynamoDbClient(DynamoDbProperties properties,
+												AwsCredentialsProvider credentialsProvider,
+												AwsRegionProvider regionProvider) throws IOException {
+			DaxProperties daxProperties = properties.getDax();
+
+			software.amazon.dax.Configuration.Builder configuration = toAwsDaxConfiguration(daxProperties);
 
 			configuration.region(AwsClientBuilderConfigurer.resolveRegion(properties, regionProvider))
 					.credentialsProvider(credentialsProvider).url(properties.getDax().getUrl());
 			return ClusterDaxClient.builder().overrideConfiguration(configuration.build()).build();
 		}
 
+		@ConditionalOnMissingBean
+		@Bean
+		public DynamoDbAsyncClient daxDynamoDbAsyncClient(DynamoDbProperties properties,
```

Comment:
Thanks for PR @driverpt!

You sneaked in async client ;-) Is there a particular reason for that? 

As much as we do want to autoconfigure async clients we need to think how to do it consistently in every module so that we don't create beans that consume resources without users knowing about it or needing it.

One of the principles of 3.0 was to not create more than needed in contract to 2.x, so we need to be careful.

Reply:
Because in the project we just upgraded to Spring Boot 3, we use WebFlux + Async Clients :)

I think it's harmless because it has `@ConditionalOnBean`